### PR TITLE
Blog Page FE

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1347772119)

### Description of Changes Made

- Adds FE styles for the `blog_detail` page including
- - Author(s)
- - Title, tags, read time
- - Related posts

Everything on this page is shifted over by 1 column in the design compared to the standard page to make space for the authors in the sidebar so i've used the template class modifier for that. The template class formatting was in BEM for some reason so i've updated that.

I've also added the spacing variables to tailwind so we can use them in utility classes - I've added a note to keep them in sync with the vars in `_variables.scss` should they ever change. 

### How to Test

Can be tested on live blog pages if you have staging data. 

Alternatley in the PL here: [http://localhost:8000/pattern-library/render-pattern/patterns/pages/blog/blog_detail.html](http://localhost:8000/pattern-library/render-pattern/patterns/pages/blog/blog_detail.html)

### Screenshots

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-01-19 at 14 32 25

Screenshot 2024-01-19 at 14 32 38

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [x] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [x] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] New font variants have not been added
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/base.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% load static wagtailuserbar wagtailcore_tags wagtailimages_tags navigation_tags util_tags wagtailaccessibility_tags %}
 <!doctype html>
-<html class="{% block body_class %}template__{{ page.get_verbose_name|slugify }}{% endblock %}" lang="en-GB">
+<html class="{% block body_class %}template-{{ page.get_verbose_name|slugify }}{% endblock %}" lang="en-GB">
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/base.html:3]: I'm not sure why this was in BEM format - could just be leftover from the current site but updated to how we do it in WK. (I can't find anything in the codebase for `template__`)

File: tailwind.config.js [hunk 2]
```diff
@@ -24,10 +24,24 @@ module.exports = {
         screens: {
             md: '599px',
             lg: '1023px',
+            xlg: '1280px',
```
Comment (chris-lawton) [inline tailwind.config.js:27]: Matching the breakpoints set in `_variables.scss`

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html [hunk 3]
```diff
@@ -12,7 +12,7 @@ <h2 class="listing-avatar__title listing__title">{{ title }}</h2>
                     <span class="listing__name mr-1.5">{{ name }}</span>
                 </p>
                 <p class="listing__meta">
-                    <span class="mr-1.5">{{ page_type }},</span> {{ reading_time }}
+                    <span class="mr-1.5">{{ page_type }},</span> {{ reading_time }} min read
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html:15]: It looks like the `reading_time` and `read_time` vars (latter used on `blog_detail.html`) only output the number, so I needed to add the 'min read' text here.

File: tbx/static_src/sass/components/_avatar.scss [hunk 4]
```diff
@@ -2,17 +2,19 @@
...
-    width: 72px;
-    height: 72px;
+    width: $avatar-size-small;
+    height: $avatar-size-small;
 
     @include media-query(medium) {
-        width: $avatar-size-desktop;
-        height: $avatar-size-desktop;
+        width: $avatar-size-large;
+        height: $avatar-size-large;
```
Comment (chris-lawton) [inline tbx/static_src/sass/components/_avatar.scss:10]: Renamed this and added two others which should be enough to cover all sizes in the design (If not this naming convention will allow us to add more).

File: tbx/project_styleguide/templates/patterns/atoms/see-more/see-more.html [hunk 5]
```diff
@@ -1,11 +1,11 @@
 {% if text %}
     {% if fetch_data %}
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/atoms/see-more/see-more.html:2]: I think we can totally ditch this `fetch_data` logic - I think it was related to the js load more functionality

File: tbx/project_styleguide/templates/patterns/atoms/tag/tag.html [hunk 6]
```diff
@@ -1,6 +1,6 @@
-<a  class="tag"
-    href="{{ tag_link_base }}?filter={{ tag_slug }}"
+<a class="tag"
+    href="{{ tag_link_base }}?filter={% firstof item.slug tag_slug %}"
     {% if is_active %}aria-current="page" {% endif %}
 >
-    {% if has_all_filter %}All{% else %}{{ tag_name }}{% endif %}
+    {% if has_all_filter %}All{% else %}{% firstof item.name tag_name %}{% endif %}
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/atoms/tag/tag.html:5]: I changed this just on the principle of trying to keep var names in included templates 'flat' as then it is easier to re-use them - is it possible to change how tag.html is included in order to pass through `tag_slug` and `tag_name` separately?

File: tbx/project_styleguide/templates/patterns/molecules/author/author.html [hunk 7]
```diff
@@ -1,17 +1,24 @@
...
-    {{ author.role }}
-{% endif %}
+    <div>
+        {% if author.person_page %}
+            <p class="author__name">
+                <a class="author__link" href="{% pageurl author.person_page %}" aria-label="View {{ author.name }}'s team profile page">
+                    {{ author.name }}
+                </a>
+            </p>
+        {% else %}
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/author/author.html:14]: Nice - this should allow us to have people who have left be authors I think? cc @engineervix

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html (hunk 3 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html:15]: You might also want to use the `pluralize` filter as I did on the blog listing.

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html [hunk 8]
```diff
@@ -26,7 +26,7 @@ <h2 class="listing-image__title listing__title">{{ title }}</h2>
                         <span class="listing__job-title">{{ job_title }}</span>
                     </p>
                     <p class="listing__meta">
-                        <span class="mr-1.5">{{ page_type }},</span> {{ reading_time }}
+                        <span class="mr-1.5">{{ page_type }},</span> {{ reading_time }} min read
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html:29]: same here

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html [hunk 9]
```diff
@@ -23,35 +23,52 @@
...
+                {% endfor %}
+            </div>
+        </div>
 
-        {% if page.read_time %}
-            / {{ page.read_time }} min read
-        {% endif %}
-    </div>
+        <div class="blog-meta">
+            <span>Insight</span>
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:47]: There was a lot of discussion around the use of 'Insights' for blog pages - I don't think it was finally decided. I noticed that Jeth queried if this was even needed in the design. Might be best to remove it.

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html (hunk 3 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html:15]: actually I've just realised I changed this same file - and also added the date, so you might just be able to remove these changes.

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html [hunk 10]
```diff
@@ -23,35 +23,52 @@
...
+        {% include_block page.body %}
 
-        {% for post in page.related_blog_posts %}
-            {% include "patterns/molecules/blog-item/blog-item.html" with item=post %}
-        {% endfor %}
-    {% endif %}
+        {% if page.related_blog_posts %}
+            {% include "patterns/atoms/section-title/section-title.html" with title="More" aria_label="Related blog posts" classes="section-title--related-posts" %}
 
+            <ul class="streamfield__related-posts">
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:63]: We may want to move this to an include as we will also have a streamfield block that looks very simiar - but maybe we can do that next sprint.

**Review by helenb — COMMENTED** [review]
Thanks @chris-lawton looking great - just a few points to check.

File: tailwind.config.js (hunk 2 above)
Comment (helenb) [reply tailwind.config.js:27]: great idea to update the tailwind spacing

**Review by helenb — APPROVED** [review]
Approving so you can merge when you've checked the points above

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html (hunk 3 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html:15]: I'll remove my changes and pick yours up when you merge your branch in that case

Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html]: Reset the changes in `listing--avatar.html` and `listing--image.html`

Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html]: https://github.com/torchbox/torchbox.com/pull/41/commits/6b50c4b6f607c5446e86144dd761b653bc824e75

File: tbx/project_styleguide/templates/patterns/atoms/see-more/see-more.html (hunk 5 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/atoms/see-more/see-more.html]: Removed in https://github.com/torchbox/torchbox.com/pull/41/commits/9a5fc55cb8a3ea40c37cd123f9947d8b1d83e1f7

File: tbx/project_styleguide/templates/patterns/atoms/tag/tag.html (hunk 6 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/atoms/tag/tag.html]: Thanks, I must have been in autopilot! 

Updated in https://github.com/torchbox/torchbox.com/pull/41/commits/223de523acc88a4f83d8eaa5e1ca90ad8b6ec50f (also removed `small=True` which wasn't doing anything)

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html (hunk 9 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html]: Removed in https://github.com/torchbox/torchbox.com/pull/41/commits/43d701ba3377531ee43ff0bc195ac26e0c9e38db. 

(currently means that `page_type` doesn't output anything on this branch but can see you've updated on https://github.com/torchbox/torchbox.com/pull/40/files#diff-7df244d38c599e323ae9fcdba40adb28c16115df2d60e576a6ca22ee612eec8fL15

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html (hunk 3 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html]: I'll update listing--image.html in my branch as well - had previously only changed listing--avater

**Pull request merged**